src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -492,6 +557,81 @@ private void StartWriting(byte[] buffer, int offset, int count, AsyncProtocolReq
             }
         }
 
+        private async Task StartWritingAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            // We loop to this method from the callback.
```

Comment:
Nit: Most of this is copy/paste from `StartWriting`. It would be nice to refactor and reuse (extract method) the common wrapper code and only keep different code in two separate functions. I saw only 3 parts that are different (maybe some sort of template method pattern would work here?)